### ISSUE TITLE
Take care of relative paths for files produced for release

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/AutoDoc#103]
Package authors who are preparing a release of their package need to compile the manual for the package. For example, my `release` script does that by invoking the `makedoc.g` file (if present).

There is one problem, though: If there are references in the package manual to other manuals (e.g. the GAP reference manual), then these need to be turned into URLs somehow. By default, what AutoDoc does is to invoke GAPDoc in such a way that it ends up inserting absolute file path URLs (I think). What we should do is to instead insert relative paths. Of course the relative path is potentially different everywhere. So the best what one can do (and I think that's the (in?)official rule being followed by some other packages) is to assume that packages are installed into `$GAPROOT/pkg/`, and use a relative path of `../..`.

For this, it would be super-useful if one could tell AutoDoc to do that (by passing the right option to `MakeGAPDocDoc` resp. `AutoDoc_MakeGAPDocDoc_WithoutLatex`).

Of course when I manually run `makedoc.g`, I am fine with the absolute paths, and in fact may prefer them over a (possibly wrong) relative path. So this really should be an option. And I'd suggest that this is an option that really should be enabled by the `ValueOption` machinery. This way, I can trigger it from the `release` tool w/o anybody having to modify their `makedoc.g` script...

**Comment by fingolfin** [comment]
So I wrote some code for this in d00c12b53da2df8ca82e82049d4679cfdeab3d7d but it is undocumented, and not really the final answer.

@sebasguts also suggested that since GAPDoc calls this value `gaproot`, we should use the same name.

Personally, I don't really like `gaproot`, as it doesn't make clear the purpose. OTOH, one could indeed use a non-relative path for it, if one wanted to... and finally, "relativePath" is indeed also not a very descriptive name.

Sebastian also suggested that one should be able to set this option via the options record, too. Fine by me! And of course such a version should also be documented then, etc.

But for me, it is kind of important that there is a way to set this *isolated* from other options, so that I can override its value in the `release` script. As this is my primary use case, I only made this undocumented version of it.

So, in short, I am all for making a more generic, and properly named and documented version of this flag -- but for the time being, I'd like to also retain the undocumented `relativePath`, which does *exactly* what I need. If we develop a "proper" alternative (@sebasguts, you are most welcome to do so), then I'd also switch the release tool to use that.

**Comment by sebasguts** [comment]
I did something about this in 5e23a81. Not documentated and basically the same as you did, with the differences I mentioned.

**Comment by fingolfin** [comment]
Sorry, I overlooked it! So this looks good in principle. But of course it now needs to be rebased atop my changes, and is missing documentation, too.

**Comment by sebasguts** [comment]
I will do this if we agree on a name. ```relativePath```? Going with the other names I would then suggest ```relative_path```.

**Comment by fingolfin** [comment]
well, I am also fine with the `gap_root_path` you used -- my objection was strictly to the name `gap_root`.

Of course it could also be `gap_root_rel_path` or `gap_root_relative_path`, or ... make your choice.

Just please leave in the `relativePath` HACK for now, so that my release script can take advantage of it. We can mark it with a `# HACK` comment or whatever, and explain why it is still there (for now, at least)

**Comment by sebasguts** [comment]
Alright.
